typing/ctype.ml

```diff
@@ -2521,6 +2532,31 @@ let constrain_type_jkind ~fixed env ty jkind =
                   (Not_a_subjkind (ty's_jkind, jkind, sub_failure_reasons)))
              end
           in
+          let or_null ~fuel ty is_open modality =
+            let error =
```

Comment:
OCaml is strict! So this structure gets allocated unconditionally, even on the happy path. Boo. Better to make this into a thunk (e.g. `let error () = ...`).

Reply:
Fixed.